csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/AssemblyCacheExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    internal static class AssemblyCacheExtensions
+    {
+        private static readonly Version emptyVersion = new Version(0, 0, 0, 0);
+
+        /// <summary>
+        /// This method orders AssemblyInfos by version numbers (.net core version first, then assembly version). Finally, it orders by filename to make the order deterministic.
+        /// </summary>
+        public static IOrderedEnumerable<AssemblyInfo> OrderAssemblyInfosByPreference(this IEnumerable<AssemblyInfo> assemblies, IEnumerable<string> frameworkPaths)
+        {
+            // prefer framework assemblies over others
+            bool initialOrdering(AssemblyInfo info) => frameworkPaths.Any(framework => info.Filename.StartsWith(framework, StringComparison.OrdinalIgnoreCase));
+
+            var ordered = assemblies is IOrderedEnumerable<AssemblyInfo> o
+                ? o.ThenBy(initialOrdering)
+                : assemblies.OrderBy(initialOrdering);
+
+            return ordered
+                .ThenBy(info => info.NetCoreVersion ?? emptyVersion)
+                .ThenBy(info => info.Version ?? emptyVersion)
```

Comment:
@hvitved Do you remember why we've decided to compare .net core versions before assembly versions?

Reply:
Not sure, I think it was mostly about removing duplicates.